src/FormattingTrait.php

```diff
@@ -259,23 +259,19 @@ public function toQuarter(bool $range = false): int|array
     /**
      * Returns the quarter range
      *
-     * @return array<string> Array with start and end date of quarter in Y-m-d format
+     * @return array{0: string, 1: string} Array with start and end date of quarter in Y-m-d format
      */
     public function toQuarterRange(): array
     {
         $quarter = (int)ceil((int)$this->format('m') / 3);
         $year = $this->format('Y');
 
-        switch ($quarter) {
-            case 1:
-                return [$year . '-01-01', $year . '-03-31'];
-            case 2:
-                return [$year . '-04-01', $year . '-06-30'];
-            case 3:
-                return [$year . '-07-01', $year . '-09-30'];
-            default:
-                return [$year . '-10-01', $year . '-12-31'];
-        }
+        return match ($quarter) {
+            1 => [$year . '-01-01', $year . '-03-31'],
+            2 => [$year . '-04-01', $year . '-06-30'],
+            3 => [$year . '-07-01', $year . '-09-30'],
+            default => [$year . '-10-01', $year . '-12-31'],
```

Comment:
```suggestion
            4 => [$year . '-10-01', $year . '-12-31'],
```
`$this->format('m')` is always '01' to '12'.
`$quarter` is always 1 to 4.

Reply:
I wonder if phpstan requires a default catch all?
If so, maybe it should throw a invalid argument exception.

Reply:
phpstan often complains about missing `default` and the existing code was also using default case instead of `4`, hence copilot used the same. I'll see if using `4` keeps phpstan happy.

Reply:
Do annotating `$quarter` as `int<1, 4>` makes PHPStan happy? 
Adding impossible `default` just to make PHPStan happy is not what should be done when finite list is known.

Reply:
Done